services/core-web/src/components/common/DocumentTable.tsx

```diff
@@ -231,20 +270,207 @@ export const DocumentTable = ({
+
+    setProjectTitle(rowSelection[0].project_title);
+    if (documentManagerGuids.length === 0) {
+      setTimeout(() => {
+        notification.warning({
+          key: documentTypeCode,
+          className: `progressNotification-${documentTypeCode}`,
+          message: "Error starting file compression",
+          description:
+            "Only archived files or previous document versions were selected. To download \
+          them you must go to the archived documents view or download them individually.",
+          duration: 15,
+          placement: "topRight",
+          onClose: handleCloseCompressionNotification,
+          top: 85,
+        });
+      }, 2000);
+    } else {
+      props.documentsCompression(mineGuid, documentManagerGuids).then((response) => {
+        const taskId = response.data && response.data.task_id ? response.data.task_id : null;
+        if (!taskId) {
+          setTimeout(() => {
+            notification.warning({
+              key: documentTypeCode,
+              className: `progressNotification-${documentTypeCode}`,
+              message: "Error starting file compression",
+              description: "An invalid task id was provided",
+              duration: 10,
+              placement: "topRight",
+              onClose: handleCloseCompressionNotification,
+              top: 85,
+            });
+          }, 2000);
+        } else {
+          const documentTypeIdentifier = `.progressNotification-${documentTypeCode}`;
+          const notificationElement = document.querySelector(documentTypeIdentifier);
+          const notificationPosition = notificationElement.getBoundingClientRect();
+          setNotificationTopPosition(notificationPosition.top);
+          progressRef.current = true;
+          setCompressionProgressVisible(true);
+          const poll = async () => {
+            const { data } = await props.pollDocumentsCompressionProgress(taskId);
+            if (data.progress) {
+              setCompressionProgress(data.progress);
+            }
+
+            if (data.state !== "SUCCESS" && progressRef.current) {
+              setTimeout(poll, 2000);
+            } else {
+              setFileToDownload(data.success_docs[0]);
+              setReadyForDownloadModalVisible(true);
+            }
+          };
+
+          poll();
+        }
+      });
+    }
+  };
+
+  const items: MenuProps["items"] = [
+    {
+      key: "0",
+      icon: <DownloadOutlined />,
+      label: (
+        <button
+          type="button"
+          className="full add-permit-dropdown-button"
+          onClick={() => {
+            setBeginCompressionModalVisible(true);
+          }}
+        >
+          <div>Download File(s)</div>
+        </button>
+      ),
+    },
+  ];
+
+  const rowSelectionObject = {
+    onChange: (selectedRowKeys: React.Key[], selectedRows: any) => {
+      handleRowSelectionChange(selectedRows);
+    },
+  };
+
+  return (
```

Comment:
There's a loooooot of if/ternary logic going on in the return statement here and I'm really not following it. Can we have a function or two to generate the elements for the table instead?

Reply:
Extracted the if/ternary logic out of the return statement here, and moved it into two separate functions. One for generating the bulk action html elements, and another function for generating the CoreTable